IBM Music Feature Card breaks building process in imfc.cpp:99:15:

### Describe the bug

Hello.

I tried to build dosbox-x from git commit 0d4f7ac and I got this error line.

```
imfc.cpp:99:15: error: format not a string literal and no format arguments [-Werror=format-security]
   99 |         printf((format + "\n").c_str(), args...);
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

gcc used: 13.1.1 on Archlinux.

### Steps to reproduce the behaviour

Just try to build commit 0d4f7ac using gcc 13.1.1.

### Expected behavior

Build process going on.

### What operating system(s) this bug have occurred on?

Archlinux

### What version(s) of DOSBox-X have this bug?

commit 0d4f7ac924971eb2b304c22f77feb1d3b45134f8

### Used configuration

_No response_

### Output log

_No response_

### Additional information

_No response_

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

--- a/src/hardware/imfc.cpp
+++ b/src/hardware/imfc.cpp
@@ -95,9 +95,9 @@ SDL_mutex* m_loggerMutex;
 template <typename... Args>
 void IMF_LOG(std::string format, Args const&... args)
 {
-	SDL_LockMutex(m_loggerMutex);
-	printf((format + "\n").c_str(), args...);
-	SDL_UnlockMutex(m_loggerMutex);
+	/*SDL_LockMutex(m_loggerMutex);
+	printf((format + "\n").c_str(), args...); // Causes "error: format not a string literal and no format arguments [-Werror=format-security]" on GCC 13.1.1
+	SDL_UnlockMutex(m_loggerMutex);*/
 }
 
 inline uint8_t leftRotate8(uint8_t n)

